docs/guides/integration-mobile-client-side.md

```diff
@@ -0,0 +1,628 @@
+
+If
+
+#### For Android:
+
+UID2Manager.getInstance().getAdvertisingToken()
+
+#### For iOS:
+UID2Manager.shared.getAdvertisingToken()
+
+returns null, there was no identity or valid token generated for various reasons and you will need to check if there are any errors from the generateIdentity call (read Enable Logging below) or the identity generated has expired and you need to call generateIdentity again. Read the next section to learn when to do so.
+
+## Best Practice on When to Pass DII to the UID2 Mobile SDK
+
+When this is the first time the app is open with no identity, to start the token generation call you'll need to call:
+
+#### For Android:
+
+UID2Manager.getInstance().generateIdentity()
+
+#### For iOS:
+
+UID2Manager.shared.generateIdentity()
+
+method with DII. Once an identity is generated, the advertising token (UID2 token) that you would send to the bid stream will be available upon the successful completion of the generateIdentity function.
+
+If the UID2 identity stored in local file storage has expired and cannot be refreshed, you must call
+
+#### For Android:
+
+UID2Manager.getInstance().generateIdentity()
+
+#### For iOS:
+UID2Manager.shared.generateIdentity()
+
+again to generate a new identity and resulting token. The exception is if
+
+#### For Android:
+
+UID2Manager.getInstance().getIdentityStatus()
+
+#### For iOS:
+UID2Manager.shared.identityStatus
+
+return OPT_OUT status and means the DII has opted out of UID2 and no identity/token should be generated for it.
+
+The best way to determine if DII is required by the UID2 Mobile SDKs is to always call
+
+#### For Android:
+UID2Manager.getInstance().getAdvertisingToken()
+
+#### For iOS:
+UID2Manager.shared.getAdvertisingToken()
+
+on application startup/resumption, if it returns null, then you should call
+
+#### For Android:
+UID2Manager.getInstance().generateIdentity()
+
+#### For iOS:
+UID2Manager.shared.generateIdentity()
+
+and pass DII into it again.
+
+## Opt-Out Handling
+
+If the DII provided to the
+
+#### For Android:
+UID2Manager.getInstance().generateIdentity()
+
+#### For iOS:
+UID2Manager.shared.generateIdentity()
+
+has opted out, this method will return null. Calling:
+
+#### For Android:
+UID2Manager.getInstance().getIdentityStatus()
+
+#### For iOS:
+UID2Manager.shared.identityStatus
+
+returns OPT_OUT status.
+
+In this case you should avoid repeatedly make generateIdentity() calls again as UID2 token will not be generated because the DII has opted out of UID2.
+
+## Enable Logging
+
+The UID2 Mobile SDK may generate logs which could help debugging any issues during UID2 integration work. You can enable the logging by: 
+
+#### For Android:
+during UID2Manager initialization:
+ 
+UID2Manager.init(
+
+  context = this,
+
+  isLoggingEnabled = true
+
+)
+
+#### For iOS:
+
+`UID2Settings.shared.isLoggingEnabled = true`
+
+(**GWH__SW in the Word doc the para below is in code style. Just confirming it's not meant to be part of the code?**)
```

Comment:
i think the para below can be treated as comments in code

Reply:
Done per meeting, thx.